Sources/Testing/Test.swift

```diff
@@ -183,7 +250,7 @@ extension Test {
       case name
       case displayName
       case sourceLocation
-      case testCases
+      case testCaseState
```

Comment:
I think in order to encode a test snapshot, we need to have the full set of test cases evaluated already, in which case this can just be `testCases` again.

Regardless, the keys in the snapshot encoding should generally mirror the public (or SPI) property names, not their backing storage.